State.py

```diff
@@ -60,6 +60,10 @@ def has_all_of(self, items):
         return all(map(self.prog_items.__contains__, items))
 
 
+    def count_of(self, items):
+        return len(list(filter(self.prog_items.__contains__, items)))
```

Comment:
`sum(map` instead of `len(list(filter`

Reply:
Isn't `sum` intended only for numeric values? I feel like using it on a map of booleans might work in current versions of Python but could very well break later.

Reply:
`bool` is a numeric type, but you're right that it might break later.